src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -219,6 +217,22 @@ public void clusterChanged(ClusterChangedEvent event) {
 
             // check and write changes in indices
             for (IndexMetaData indexMetaData : newMetaData) {
+
+                boolean shardsAllocatedOnThisNodeInLastClusterState = true;
+                if (isDataOnlyNode(state)) {
+                    boolean shardsCurrentlyAllocatedOnThisNode = shardsAllocatedOnLocalNode(state, indexMetaData);
```

Comment:
can we replace this entire block with just this:
```
 if (indicesService.hasIndex(indexMetaData.index())) {
                        
}
```

and drop the entire delete logic here I think we can just fix the writing here. If we don't wipe the stuff correctly that's a different issue.